frontend/src/app/pages/admin/search/admin-search.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, OnInit } from '@angular/core';
+import { Patient } from '@app/model/Patient';
+import { AlertService } from '@app/services/alert/alert.service';
+import { SearchHistoryService } from '@app/services/search-history/search-history.service';
+import { PatientService } from '@app/services/patient/patient.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-search-patient',
+  templateUrl: './admin-search.component.html',
+  styleUrls: ['./admin-search.component.scss']
+})
+export class AdminSearchComponent implements OnInit {
+
+  public personalNumber?: string;
+  public patient?: Patient;
+
+  public loading: boolean = false;
+  public submitted: boolean = false;
+
+  constructor(private _alertService: AlertService,
+              private _router: Router,
+              private _searchHistoryService: SearchHistoryService,
+              private _patientService: PatientService) {
+  }
+
+  ngOnInit(): void {
+  }
+
+  get searchHistory(): string[] {
+    return this._searchHistoryService.searchHistory;
+  }
+
+  public async onSubmit(): Promise<void> {
+    if (!this.personalNumber) {
+      return;
+    }
+
+    await this.findPatient(this.personalNumber);
+  }
+
+  public async findPatient(personalNumber: string): Promise<void> {
+    this.submitted = true;
+    this.loading = true;
+
+    try {
+      this.patient = await this._patientService.findPatientByPersonalNumber(personalNumber);
+      this._searchHistoryService.saveSearch(personalNumber);
```

Comment:
kdyz to nenajde pacianta, tak tohle uz se nemusi volat.

Reply:
A vlastne to hledani pacienta se nemusi vubec volat - protoze i pri volani historie na neexistujici cislo by to melo failnout.

Reply:
kdyz to nenajde pacienta, spadne to do catch kvuli 404

Reply:
takze se neexistujici cislo do historie neulozi